Feature/activatable actors

Frickerson opened the pull request.

Commit:
Implement activatable actor class.

Commit:
Add some MP stuff

Add authority checks.
Add bool to activate on clients or not.

TheEmidee commented on Source/GameBaseFramework/Private/Gameplay/GBFActivatableActor.cpp:
```diff
@@ -0,0 +1,66 @@
+
+    bActivated = true;
+
+    ReceiveActivate();
```
I would call OnRep_Activated

TheEmidee commented on Source/GameBaseFramework/Private/Gameplay/GBFActivatableActor.cpp:
```diff
@@ -0,0 +1,66 @@
+
+    bActivated = false;
+
+    ReceiveDeactivate();
```
same

TheEmidee commented on Source/GameBaseFramework/Private/Gameplay/GBFActivatableActor.cpp:
```diff
@@ -0,0 +1,66 @@
+{
+    Super::GetLifetimeReplicatedProps( OutLifetimeProps );
+
+    DOREPLIFETIME( AGBFActivatableActor, bActivated );
```
I would call DOREPLIFETIME_CONDITION with `bAllowActivationOnClients` as the condition

TheEmidee commented on Source/GameBaseFramework/Private/Gameplay/GBFActivatableActor.cpp:
```diff
@@ -0,0 +1,66 @@
+
+void AGBFActivatableActor::OnRep_Activated()
+{
+    if ( !bAllowActivationOnClients )
```
this should not become necessary if you add DOREPLIFETIME_CONDITION

Commit:
PR review

Add replication condition based on boolean.
Call OnRep_Activated instead of the Receive functions.
